CDP4WebServices.API/Services/ChangeLog/ChangeLogService.cs

```diff
@@ -0,0 +1,1311 @@
+
+                    if (parameterSubscriptionValueSetParameterSubscription != null)
+                    {
+                        var newCustomData = this.GetCustomAffectedThingsData(transaction, partition, rootThing, updateOperation, parameterSubscriptionValueSetParameterSubscription);
+                        customData.AddFrom(newCustomData);
+                    }
+
+                    break;
+                }
+            }
+
+            return customData;
+        }
+
+        /// <summary>
+        /// Tries to get the <paramref name="updatedThing"/>'s container.
+        /// </summary>
+        /// <param name="transaction">
+        /// The current <see cref="NpgsqlTransaction"/> to the database.
+        /// </param>
+        /// <param name="partition">
+        /// The database partition (schema) where the requested resource is stored.
+        /// </param>
+        /// <param name="updatedThing">
+        /// The updated <see cref="Thing"/>
+        /// </param>
+        /// <returns>
+        /// <paramref name="updatedThing"/>'s container <see cref="Thing"/> if found, otherwise null
+        /// </returns>
+        private Thing GetContainer(NpgsqlTransaction transaction, string partition, Thing updatedThing)
+        {
+            try
+            {
+                var containerClassType = ContainerPropertyHelper.ContainerClassName(updatedThing.ClassKind);
+
+                if (containerClassType == updatedThing.ClassKind.ToString())
+                {
+                    return null;
+                }
+
+                var containerPropertyName = ContainerPropertyHelper.ContainerPropertyName(updatedThing.ClassKind);
+                var securityContext = new RequestSecurityContext { ContainerReadAllowed = true };
+                var service = this.ServiceProvider.MapToPersitableService(containerClassType);
+
+                if (service == null)
+                {
+                    return null;
+                }
+
+                var containerPartition = partition;
+
+                if (!EngineeringModelContainmentClassType.ClassKindArray.Contains((ClassKind) Enum.Parse(typeof(ClassKind), containerClassType)))
+                {
+                    containerPartition = partition.Replace("EngineeringModel", "Iteration");
+                }
+
+                var possibleContainers = service.GetShallow(transaction, containerPartition, null, securityContext).ToList();
+
+                if (!possibleContainers.Any())
+                {
+                    return null;
+                }
+
+                var addContainer = this.GetContainerFromPossibleContainers(updatedThing, possibleContainers, containerPropertyName);
+
+                return addContainer;
+            }
+            catch (Exception ex)
+            {
+                // Ignore
+                Logger.Error(ex);
```

Comment:
what kind of error are you expecting, it is being logged but we continue anyway?

Reply:
I'm not expecting any errors, so indeed it is being logged but we continue anyway.
This particular piece of code finds the changed Thing's container only to add some information to the ChangeDescription.
I don't want an unexpected failure here to be the reason a user cannot make a specific change at all.